app/views/partials/foyer/ressources/year-moins-2.html

```diff
@@ -4,42 +4,57 @@
     Ces informations se trouvent sur votre déclaration de revenus d'impôts {{ yearMoins2 }}.
     <br>Vous pouvez la retrouver en ligne sur <a href="http://www.impots.gouv.fr/">impots.gouv.fr</a>.
   </p>
-  <form class="form-horizontal" novalidate ng-submit="submit()">
+  <form class="form-horizontal" novalidate name="form" ng-submit="submit(form)">
     <div class="row">
       <div class="col-md-12" ng-repeat="individuRef in individuRefsToDisplay">
         <h2>{{ individuRef.label }}</h2>
-        <div class="form-group" ng-repeat="rnc in individuRef.rnc">
-          <label class="control-label col-sm-3" for="{{ rnc.categorie.id }}-{{ individuRef.individu.role }}">
+        <div
+          class="form-group"
+          ng-repeat="rnc in individuRef.rnc"
+          ng-init="inputName = rnc.categorie.id +  '-' + individuRef.individu.role"
+          ng-class="{ 'has-error': form.$submitted && form[inputName].$invalid }"
+          >
+          <label class="control-label col-sm-3" for="{{ inputName }}">
             {{ rnc.categorie.label }}
           </label>
           <div class="col-sm-3">
             <span class="input-group">
               <input
                 type="number"
-                id="{{ rnc.categorie.id }}-{{ individuRef.individu.role }}"
+                id="{{ inputName }}"
+                name="{{ inputName }}"
                 class="form-control text-right"
                 ng-model="rnc.montant"
                 min="0"
                 select-on-focus
-                zero-to-empty>
+                zero-to-empty
+                aria-describedby="help-{{ inputName }}">
               <span class="input-group-addon hidden-xs">€</span>
             </span>
+            <ul class="text-danger" ng-if="form.$submitted && form[inputName].$invalid">
+              <li>
+                Montant invalide
+              </li>
```

Comment:
This doesn't seem to use exactly the same structure as other validators, so I don't see the benefits of using the `<ul><li>` construct when there is only one element and the display is controlled only with Angular.

Removing the disc and aligning to the right would probably improve the aesthetics of the error. Also, I find it strange that the error is bigger than the label.

![capture d ecran 2017-06-27 a 18 08 13](https://user-images.githubusercontent.com/222463/27597877-fe8ca93c-5b63-11e7-9f2f-c629f0638611.png)